js/main.js

```diff
@@ -0,0 +1,165 @@
+    function questionsAndTime() {
+      //Creating an animated loading bar
+      var $barRow = document.createElement('div');
+      $barRow.className = "bar";
+      $quizContainer.prepend($barRow);
+      var $inRow = document.createElement('div');
+      $inRow.className = "in";
+      $barRow.appendChild($inRow);
+      //Add timeout to animated loading bar
+      var timeBar = setTimeout(quizTimer, 10000)
+      function quizTimer() {
+        console.log('NEXT QUESTION, remake DOM');
+        var $navH1 = document.querySelector('.navh1');
+        data.wrongPokemon.push(data.currentPokemon)
+        data.currentNumber++;
+        $navH1.textContent = "Question " + data.currentNumber
+        $quizContainer.remove();
+        createQuizContainer();
+        generateFour();
+        getPokemonPicture()
+        console.log(data);
+      }
+      //Shuffle current array
+      var shuffledFour = _.shuffle(data.currentFour);
+      //Creating 4 buttons
+      for (var i = 0; i < 4; i++) {
+        var $button = document.createElement('button');
+        $button.className = "white-button justify-center";
+        $button.textContent = allPokemonList[shuffledFour[i] - 1]
+        $quizContainer.appendChild($button);
+        $button.addEventListener('click', questionClick)
+        function questionClick() {
+          //Clear timeout
+          clearTimeout(timeBar)
+          console.log(event.target.textContent);
+          if (data.currentNumber === 10) {
+            alert('goodjob')
+          } else if (event.target.textContent === data.currentPokemon) {
+            console.log('NEXT QUESTION, remake DOM');
+            var $navH1 = document.querySelector('.navh1');
+            var $dots = document.querySelectorAll('.col-tenth')
+            $dots[data.currentNumber - 1].textContent = ""
+            var $icon = document.createElement('img');
+            $icon.className = "icon"
+            $icon.setAttribute('src', 'images/pokeball.png')
+            $dots[data.currentNumber - 1].appendChild($icon)
+            //Push current pokemon
+            data.correctPokemon.push(data.currentPokemon);
+            data.currentNumber++;
+            $navH1.textContent = "Question " + data.currentNumber
+            $quizContainer.remove();
+            createQuizContainer()
+            generateFour();
+            getPokemonPicture();
+            console.log(data);
+          } else {
+            //WRONG ANSWERS
+            console.log('NEXT QUESTION, remake DOM');
+            var $navH1 = document.querySelector('.navh1');
+            data.wrongPokemon.push(data.currentPokemon)
+            data.currentNumber++;
+            $navH1.textContent = "Question " + data.currentNumber
+            $quizContainer.remove();
+            createQuizContainer();
+            generateFour();
+            getPokemonPicture()
+            console.log(data);
+          }
+        }
+      }
+    }
+  }
```

Comment:
I **_sincerely recommend_** you don't create function definitions in function definitions like this. You should create all function definitions in the global scope. If you are doing this because you need a piece of data in a certain scope, that means you actually need to use parameters/arguments for that function.